onnxruntime/core/providers/cuda/generator/random.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/cuda/cuda_kernel.h"
+
+#include "core/providers/cuda/generator/random_impl.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+#define RANDOM_COMPUTE_IMPL(name)                                                                        \
+  template <typename T>                                                                                  \
+  struct name##ComputeImpl {                                                                             \
+    void operator()(const cudaDeviceProp& prop, cudaStream_t stream, const int64_t N, const float alpha, \
+                    const float beta, PhiloxGenerator& generator, Tensor& Y) const {                     \
+      typedef typename ToCudaType<T>::MappedType CudaT;                                                  \
+      CudaT* Y_data = reinterpret_cast<CudaT*>(Y.template MutableData<T>());                             \
+      name##KernelImpl<CudaT>(prop, stream, N, alpha, beta, generator, Y_data);                          \
+    }                                                                                                    \
+  };
+
+RANDOM_COMPUTE_IMPL(RandomNormal)
+RANDOM_COMPUTE_IMPL(RandomUniform)
+
+#undef RANDOM_COMPUTE_IMPL
+
+class RandomBase : public CudaKernel {
+ protected:
+  RandomBase(const OpKernelInfo& info) : CudaKernel(info) {
+    float seed = 0.f;
+    if (info.GetAttr<float>("seed", &seed).IsOK()) {
```

Comment:
we need to use the global seed, if the seed in attribute is not set. 
See Random CPU kernel for reference.

Reply:
When seed attr is not set, we will use default generator, which use the global seed, and it's static, so different nodes will generate different data. Dropout is also doing this way.

PhiloxGenerator& generator = generator_ ? *generator_ : PhiloxGenerator::Default();

Reply:
got it.